ledger/src/blockstore_processor.rs

```diff
@@ -337,34 +338,49 @@ fn execute_batches(
             let cost = tx_cost.sum();
             minimal_tx_cost = std::cmp::min(minimal_tx_cost, cost);
             total_cost = total_cost.saturating_add(cost);
-            cost
+            tx_cost
         })
         .collect::<Vec<_>>();
 
+    if bank
+        .feature_set
+        .is_active(&feature_set::apply_cost_tracker_during_replay::id())
+    {
+        let mut cost_tracker = bank.write_cost_tracker().unwrap();
+        for tx_cost in &tx_costs {
+            cost_tracker
+                .try_add(tx_cost)
+                .map_err(TransactionError::from)?;
```

Comment:
The feature gated change here to apply exact same cost tracking as leader did when produced the block. Fail with TransactionError should any limits were exceeded. If all blocks were produced honestly, check here should never fail, unless leader and validators running different version of cost_model/cost_tracker/block_cost_limits as mentioned in PR description.

Reply:
yeah, think so. Just a side note, removing enabled feature gate code is no small task, it should be treated at least as refactoring.